packages/lit-html/src/private-ssr-support.ts

```diff
@@ -57,6 +57,22 @@ export const _$LH = {
         return resolveOverrideFn(this, values);
       }
     },
+  patchDirectiveResolve: (
```

Comment:
We want to remove `overrideDirectiveResolve` at some point... but older versions of ssr may depend on a newer version of lit-html... when can we remove it?

Reply:
I think we get to remove it when we want to consider the current version of `@lit-labs/ssr` unsupported. Perhaps we can add a deprecation warning that logs when `overrideDirectiveResolve` is called that encourages users to update `@lit-labs/ssr`? They also should do that since there's a memory leak.

Reply:
Great comment! The positive of adding API here is that this doesn't increase `lit-html`'s size as it's only pulled from in SSR. The negative is that there is a larger API surface here.

We would want to be very slow at removing `overrideDirectiveResolve `, and potentially leave it in until most people are using Lit 3. If we want to continue down this approach I could add a comment to document this.

Alternatively, we could revert back to the very first implementation where we patch fix the memory leak from only the ssr side, without any modifications to `private-ssr-support`.

Reply:
I think one takeaway is that overrideDirectiveResolve isn't low-level enough, and we should check that patchDirectiveResolve doesn't have the same problem. Is there's a lower-level primitive we could add and move the patching itself into ssr?

Reply:
I'm quite motivated to fix the underlying issue since the leak is pretty serious. Wdyt about this very simple fix that I've split out: https://github.com/lit/lit/pull/4528

Then we can work on a much more robust fix that also handles minification correctly.